.gitlab-ci.yml

```diff
@@ -0,0 +1,78 @@
+include: "https://gitlab-templates.ddbuild.io/slack-notifier/v3-sdm/template.yml"
+
+stages:
+  - test
+  - after-test
+  - check-downstream
+  - notify
+
+.skip_stable_branches: &skip_stable_branches
+  - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+    when: never
+  - if: $CI_COMMIT_BRANCH == "datadog-5.5.0"
+    when: never
+  - if: $CI_COMMIT_BRANCH =~ /^7.[0-9]{2}.x/
+    when: never
+  - if: $CI_COMMIT_TAG != null
+    when: never
+
+trigger-agent-build:
+  rules:
+    - *skip_stable_branches
+    - when: always
+  stage: test
+  trigger:
+    project: datadog/datadog-agent
+    strategy: depend
+  variables:
+    OMNIBUS_RUBY_VERSION: $CI_COMMIT_BRANCH
+    RUN_ALL_BUILDS: "true"
+    RUN_KITCHEN_TESTS: "true"
+
+on-success:
+  rules:
+    - *skip_stable_branches
+    - when: on_success
+  stage: after-test
+  needs: ["trigger-agent-build"]
+  tags: ["arch:amd64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:v19805261-b468a29
+  script:
+    - echo success > output.pipeline
+  artifacts:
+    paths:
+      - output.pipeline
+    expire_in: 1 week
+
+# This job is monitored on github and acts as a required check.
+# A few details on the setup:
+# We can't monitor a trigger job from github, it has to have a script section.
+# We need the monitored job to fail if the downstream pipeline failed, and succeed if
+# the downstream pipeline succeeded. If the job is skipped, it isn't taken into account
+# on github.
+# This is where the on_success job kicks in. It run only if the previous job completes
+# and writes `success` in a file that's fetched by the `check-downstream-pipeline` job.
+# This allows us to have the check-downstream-pipeline to always run while replicating
+# the status of the downstream pipeline.
+check-downstream-pipeline:
+  rules:
+    - *skip_stable_branches
+    - when: always
+  stage: check-downstream
+  tags: ["arch:amd64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:v19805261-b468a29
+  when: always
+  script:
+    - cat output.pipeline
```

Comment:
For sure, but to do that in a sane way it would make most sense to make that into a library and that's a larger effort that should be a followup outside of this PR.